### DISCUSSION TITLE
Question about trailing `src/` in remapping

### DISCUSSION THREAD

**Discussion by NIC619** [sablier-labs/evm-monorepo#676] [Q&A] [answered]
Hi @PaulRBerg ! Really appreciate your Random Foundry Tips series.

I was wondering in the recent tip regarding trailing `src/` in remapping, under what circumstances will the problem manifest itself?

Does it happen when you have foundry and hardhat in the same repo? Can it be prevented by installing `foundry-hardhat`?

https://twitter.com/PaulRBerg/status/1692245880584142978

**Comment by PaulRBerg** [comment, marked as answer]
Hey @NIC619, thank you for your question.

> under what circumstances will the problem manifest itself?

I honestly can't remember off the top of my head, but it was related to a VSCode extension that wasn't able to parse the remappings in `foundry.toml`.

Generally, the issue is that some tools except the remappings to be defined in `remappings.txt`, and not anywhere else.

But if you haven't bumped into any issues yet, then it seems to be a reasonable option to keep your remappings in `foundry.toml`.

**Reply by PaulRBerg** [reply]
Did this answer your question, @NIC619?

**Reply by NIC619** [reply]
> Generally, the issue is that some tools except the remappings to be defined in `remappings.txt`, and not anywhere else.

Is it related to trailing `src/` in the remapping (regardless of in which file the remappings are defined) or generally just about where the remappings are defined?

**Reply by PaulRBerg** [reply]
The latter (IIRC)

**Reply by NIC619** [reply]
OK got it. Thanks for answering!
